### PR TITLE
Fix/leaflet map display Closes #1130

### LINKED ISSUE #1130: move actual input form into a partial template for use in location entry across the site #1070

### PR DESCRIPTION
**Pull request by mridulnagpal**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.
Closes #1130 
Thanks!

### CODE REVIEW

**Comment by mridulnagpal** [comment]
Added a div tag in helper template now there is no error for js

File: app/views/map/_leaflet.html.erb [hunk 1]
```diff
@@ -1,8 +1,10 @@
-<script>
-        var map = L.map('map').setView([<%= lat %>,<%= lon %>], 10);
-        L.tileLayer("https://a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(map)
+<div id="map">
```
Comment (jywarren) [inline app/views/map/_leaflet.html.erb]: I think we have to remove the map div from the sidebar templates too if we're going to add it here, or there'll be two maps in the sidebar. Make sense?

Comment (jywarren) [inline app/views/map/_leaflet.html.erb]: Also, I believe the div should precede the script tags, not enclose them, here.

**Comment by jywarren** [comment]
Can you make the form template into a partial, by preceding its name with an underscore and adding "render partial: 'map/form'" in the controller?

**Comment by mridulnagpal** [comment]
shouldn't it be locations/form ??

**Comment by ananyo2012** [comment]
@mridulnagpal Can you please squash your commits so that there is just one?

**Comment by mridulnagpal** [comment]
@ananyo2012 Actually I tried squashing them but there seems to be an issue with my remote, therefore @jywarren told me that he would squash the commits while merging. But if squashing is necessary i would surely give it a go again .

**Comment by ananyo2012** [comment]
yeah its not a much of an issue. But its good to know how its done so it would easier for you in future. Give it a try and if you face problems you can ask. We are there to help.

**Comment by mridulnagpal** [comment]
@ananyo2012 Sure I'll try

File: app/views/sidebar/_related.html.erb [hunk 2]
```diff
@@ -77,7 +77,7 @@
   <% if @maps && @maps.length > 0 && params[:controller] != "search" %>
 
     <br />
-    <div style="width:100%;margin-left:0;height:300px;" id="map"></div>
+    <div style="width:100%;margin-left:0;height:300px;"></div>
```
Comment (jywarren) [inline app/views/sidebar/_related.html.erb]: This does move the map div to the partial, but it leaves the styled div here, unused. Can you remove this line entirely, and move the style attributes to the partial `_leaflet.html.erb`? Thanks!

**Comment by mridulnagpal** [comment]
@jywarren Done

**Comment by mridulnagpal** [comment]
@jywarren Do I need to add anything else or is this fine??

**Comment by jywarren** [comment]
Perfect -- we'll want to test this out -- i'll merge it to stable -- in a few minutes, can you take a look at this page on our new staging server to see if it works properly?

http://testing.laboratoriopublico.org/wiki/gulf-coast

**Comment by mridulnagpal** [comment]
@jywarren Seems to work fine

**Comment by jywarren** [comment]
Super!

**Pull request merged**